Edits to links to enable single-sourcing of xip-0-purpose-process.md

jhaaaa opened the pull request:
- Proposes that we change relative links to full URLs in `xip-0-purpose-process.md` so that we can single-source this great content to the `xmtp-dot-org` repo and publish it to the Dev Portal. (Relative links break in the context of the `xmtp-dot-org` repo.)
- Updates "Standards Track" to "Standards" per a change in terms we used in the XMTP Litepaper
- Updates community forum URLs to point to the XMTP GitHub Discussions - XIPs category
- Updates title value in page header to use sentence case and "and" instead of "&" to match documentation style
- Makes a few nit edits

saulmc reviewed (approved):
love it

mkobetic reviewed (approved):
Looks good.

galligan commented:
Thanks for this @jhaaaa many of these things are great.

Would we be able to separate out the full URL linking in a separate PR? By using the full links to the content, the links no longer work in the same context when browsing within GitHub, which in the future may be valuable. Furthermore, it could create a challenging dependency whereby any future XIPs that get written may need to also see hardcoded URLs added, and subsequently changed if our `.org` site structure changes in any way.

We should be able to find a way to pull in content in a way that would actually work from a URL perspective.

jhaaaa commented on XIPs/xip-0-purpose-process.md:
```diff
@@ -1,12 +1,12 @@
-title: XIP Purpose, Process, & Guidelines
-discussions-to: https://community.xmtp.org/t/xip-0-xip-purpose-process-guidelines/475
+title: XIP purpose, process, and guidelines
+discussions-to: https://github.com/orgs/xmtp/discussions/categories/xips
```
Ah, I see - thank you for your patient reply on this @galligan! 

Regarding opening a new topic for XIP-0 in GitHub Discussions - now that I see what I missed - I don't mean to complicate things or disrupt. Maybe we can just keep pointing to https://community.xmtp.org/t/xip-0-xip-purpose-process-guidelines/475 as the `discussion-to` URL of record?

jhaaaa commented:
@galligan thank you for your feedback on this PR! 

- I've reverted the `discussions-to` URL change as I think I may have overcomplicated things in my misreading of the purpose of that URL. I think it works to keep it as-is as the URL of record!
- I reverted back to using relative paths. I'm going to adjust the xmtp.org content to just point to this XIP-0 for all of the details. I may have been trying to do too much in single-sourcing it from this location.

Please let me know if you'd like to see any other changes - much gratitude to you!

galligan commented:
> I reverted back to using relative paths. I'm going to adjust the xmtp.org content to just point to this XIP-0 for all of the details. I may have been trying to do too much in single-sourcing it from this location.

@jhaaaa I’d be happy to connect with you on this and think through how the content sourcing is going, and how we might be able to still accomplish those goals given the relative linking.

One example I’ll provide is that Ethereum has an [EIPs website](https://eips.ethereum.org), which does source content directly from a [related GitHub repo](https://github.com/ethereum/EIPs), which is itself reliant on relative links.

galligan reviewed (approved):
Looks good for now.

@jhaaaa we definitely will need to update the XIP-0 link as we will eventually be deprecating that discussion spot. So opening something in GitHub discussions would make sense.

jhaaaa commented:
Thank you so much for your feedback and understanding, @galligan!

>I’d be happy to connect with you on this and think through how the content sourcing is going, and how we might be able to still accomplish those goals given the relative linking.

Yes, please! I am confident that we can accomplish a wonderful single-sourcing solution. As I'm sure you are keenly aware =), I'm currently working on getting us to a stable v1 release state for xmtp.org as soon as possible. I have a few single-sourcing issues opened that I am deferring to after v1 and I'm adding this one to the list! I would be so grateful to get your input on a solution -- and thank you for links to the EIPs website and repo -- this is a great example to follow! More soon.
